client/src/partials/finance/cashflow/cashflow.js

```diff
@@ -0,0 +1,326 @@
+(function (angular) {
+  'use strict';
+
+  angular.module('bhima.controllers')
+  .controller('CashflowController', CashflowController);
+
+  // dependencies injection
+  CashflowController.$inject = [
+    'CashboxService', 'CashflowService', 'NotifyService', 'DateService',
+    'ExchangeRateService', 'VoucherService', 'ModalService', 'LanguageService'
+  ];
+
+  /**
+   * Cashflow controller
+   *
+   * @description
+   * This controller is responsible of cash flow report, that report include
+   * all incomes minus all depenses
+   */
+  function CashflowController(Cashbox, Cashflow, Notify, Dates, Exchange, Voucher, Modal, Languages) {
+    var vm = this, session = vm.session = {};
+
+    // global variables
+    vm.state = 'default';
+
+    // expose to the view
+    vm.showDetails = false;
+    vm.basicDate = Dates.util.str;
+    vm.getTransferText = Voucher.getTransferText;
```

Comment:
This was a good idea.  Moving functions into the service is a really good abstraction.

I have a couple remarks:
 1. You never use `getTransferText()` in the view, so you do not need to bind it to the `vm` object.
 2. The other remark I will make to you in person.